Src/IronPython/Runtime/PythonEncoding.cs

```diff
@@ -367,23 +437,29 @@ public abstract class PythonDecoderFallback : DecoderFallback, ICloneable {
         protected abstract class PythonDecoderFallbackBuffer : DecoderFallbackBuffer {
             private readonly char _marker;
             private readonly Queue<char> _fallbackChars;
+            private readonly bool _hasIndexBug;
 
             private int _fbkCnt;
             private int _charNum;
             private int _charCnt;
 
-            public PythonDecoderFallbackBuffer(bool isPass1, int encodingCharWidth)
-            {
+            public PythonDecoderFallbackBuffer(bool isPass1, PythonEncoding encoding) {
                 _marker = isPass1 ? Pass1Marker : Pass2Marker;
                 _fallbackChars = isPass1 ? null : new Queue<char>();
-                this.EncodingCharWidth = encodingCharWidth;
+                this.EncodingCharWidth = encoding.CharacterWidth;
+                this.CodePage = encoding.CodePage;
+                _hasIndexBug = encoding.HasBugCorefx29898;
```

Comment:
If `HasBugCorefx29898` can be static then we don't need `_hasIndexBug`.

Reply:
Right.